ER図とREADMEの作成

mokamokononono opened the pull request:
# what
+ ER図の作成
  + gyazoURL: https://gyazo.com/084e9795cd211e4eb946430e75e3e490
+ READMEの作成

# why
+ フリーマーケットサイトに必要なデータベース設計を行うため
  + ユーザー管理機能
  + 商品出品機能
  + 商品購入機能

mokamokononono committed:
ER図とREADMEの作成

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| Column          | Type   | Options                    |
+| --------------- | ------ | -------------------------- |
+| email           | string | null: false, unique: true  |
+| password        | string | null: false                |
```
パスワードのカラム名はencrypted_passwordにしましょう！
deviseにはパスワードを保存するカラムが標準で搭載されており、そちらを使用するためです。

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| email           | string | null: false, unique: true  |
+| password        | string | null: false                |
+| nick_name       | string | null: false, unique: true  |
+| kanji_name      | string | null: false                |
+| katakana_name   | string | null: false                |
```
本名の項目は「姓」「名」分けて保存できるようにしましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| --------------- | ------------ | ------------------------------ |
+| name            | string       | null: false                    |
+| price           | integer      | null: false                    |
+| image           | ActiveStrage | null: false                    |
```
ActiveStrageを利用するデータは記述しなくてもOKです！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| price           | integer      | null: false                    |
+| image           | ActiveStrage | null: false                    |
+| description     | text         | null: false                    |
+| category        | string       | null: false                    |
+| condition       | string       | null: false                    |
+| shipping_charge | string       | null: false,                   |
+| shipping_source | string       | null: false,                   |
+| shipping_date   | string       | null: false,                   |
```
商品の状態, 配送料の負担, 発送元の地域, 発送までの日数, カテゴリーのカラムはactive_hashで実装の予定のため、integer型にしましょう！
詳しくは下記のカリキュラムを参考にしてください！
https://master.tech-camp.in/curriculums/5945

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,64 @@
+
+| Column        | Type       | Options                        |
+| ------------- | ---------- | ------------------------------ |
+| zip_code      | integer    | null: false                    |
```
郵便番号にはハイフンが含まれるため、string型にしましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| ------------- | ---------- | ------------------------------ |
+| zip_code      | integer    | null: false                    |
+| address       | string     | null: false                    |
+| phone_number  | integer    | null: false                    |
```
integer型は「０」から始まる数字を許可できないため、電話番号はstring型にしましょう！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,64 @@
+
+| Column        | Type       | Options                        |
+| ------------- | ---------- | ------------------------------ |
+| zip_code      | integer    | null: false                    |
+| address       | string     | null: false                    |
+| phone_number  | integer    | null: false                    |
+| order         | references | null: false, foreign_key: true |
```
必要なカラムが不足しているかと存じます。
見本アプリを参考に、もう一度洗い出してみましょう！

shodai-ito reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

その後、再度レビュー依頼フォームのご提出をよろしくお願いいたします。

mokamokononono committed:
ER図とREADMEの修正

・READMEのActiveStrageカラム記述の削除
・ActiveHashに関わるカラムの型をinteger型に修正
・その他不足カラムの追加
　・氏名の苗字と名前の分離
　・配送先住所の都道府県、市区町村、番地などを分離

mokamokononono commented:
# What
- READMEの修正
  - READMEのActiveStrageカラム記述の削除
  - ActiveHashに関わるカラムの型をinteger型に修正
  - その他不足カラムの追加
    - 氏名の苗字と名前を別のカラムに分離
    - 配送先住所の都道府県、市区町村、番地などを別のカラムに分離
- ER図の修正
  -  https://gyazo.com/2a6c9ea093d33ef4d68f734166728bc6
 
# why
+ コードレビューからコードを修正し、フリーマーケットサイトに必要なデータベース設計を行うため
  + ユーザー管理機能
  + 商品出品機能
  + 商品購入機能

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,68 @@
+| name            | string       | null: false                    |
+| price           | integer      | null: false                    |
+| description     | text         | null: false                    |
+| category        | integer      | null: false                    |
+| condition       | integer      | null: false                    |
+| shipping_charge | integer      | null: false,                   |
+| shipping_source | integer      | null: false,                   |
+| shipping_date   | integer      | null: false,                   |
```
active_hashを利用するカラム名には_idを追加しておきましょう！
外部から見た際、_idがあれば何が保存されているのかわかりやすくなります！

shodai-ito commented on README.md:
```diff
@@ -1,24 +1,68 @@
+| Column        | Type       | Options                        |
+| ------------- | ---------- | ------------------------------ |
+| zip_code      | string     | null: false                    |
+| prefecture    | integer    | null: false                    |
```
こちらも_idを追記しておきましょう！

mokamokononono replied:
zip_codeは郵便番号を想定しているため、ActiveHashを用いないと考えられるため、idを追記する必要性はないと考えています。

shodai-ito reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました！
コメントしましたので、ご確認をお願いいたします！

その後、再度レビュー依頼フォームのご提出をよろしくお願いいたします。

mokamokononono committed:
ER図とREADMEの修正

- ActiveHashに関わるカラム名に_idを追記

mokamokononono commented:
# what
- READMEの修正
  - ActiveHashに関わるカラム名に_idを追記
- ER図の修正
  - ActiveHashに関わるカラム名に_idを追記
  - https://gyazo.com/547e46906c5616c2b3c70890c9dae7f4

# why
+ コードレビューからコードを修正し、フリーマーケットサイトに必要なデータベース設計を行うため
  + ユーザー管理機能
  + 商品出品機能
  + 商品購入機能

kata-myu reviewed (commented):
レビュー依頼ありがとうございます。
LGTMとさせていただきます。実装お疲れ様でした。

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き実装頑張っていきましょう。
